src/Raven.Server/Documents/Replication/DocumentReplicationTransport.cs

```diff
@@ -73,29 +75,52 @@ public class DocumentReplicationTransport : IDisposable
             return etag;
         }
 
-        //TODO : add here logic so reconnection is attempted couple of times before giving up
-        private async Task<WebSocket> GetAndConnectWebSocketAsync()
-        {
-            var uri = new Uri($"{_url?.Replace("http://", "ws://")?.Replace(".fiddler", "")}/databases/{_targetDbName?.Replace("/", string.Empty)}/documentReplication?srcDbId={_srcDbId}&srcDbName={EscapingHelper.EscapeLongDataString(_srcDbName)}");
-            try
-            {
-                if (Sparrow.Platform.Platform.RunningOnPosix)
-                {
-                    var webSocketUnix = new RavenUnixClientWebSocket();
-                    await webSocketUnix.ConnectAsync(uri, _cancellationToken);
+		//TODO : add here logic so reconnection is attempted couple of times before giving up
+		private async Task<WebSocket> GetAndConnectWebSocketAsync()
+		{
+			var uri = new Uri($"{_url?.Replace("http://", "ws://")?.Replace(".fiddler", "")}/databases/{_targetDbName?.Replace("/", string.Empty)}/documentReplication?srcDbId={_srcDbId}&srcDbName={EscapingHelper.EscapeLongDataString(_srcDbName)}");
+			try
+			{
+				if (Sparrow.Platform.Platform.RunningOnPosix)
+				{
+					var webSocketUnix = new RavenUnixClientWebSocket();					;
```

Comment:
The entity responsible for retry is the code that monitors all connections. And there should be no difference between connection aborted on start and connection aborted while talking to client.